tfc-dump.pl

```diff
@@ -30,140 +36,209 @@
 
 Getopt::Long::Configure qw(gnu_getopt);
 GetOptions(
-	'org|o=s'       => \$tfc_org_name,
-	'workspace|w=s' => \$tfc_workspace_name,
-	'all|a'         => \$all_workspaces,
-	'quiet|q'       => \$quiet_mode,
-	'help|h'        => \$help
+        'org|o=s'       => \$tfc_org_name,
+        'workspace|w=s' => \$tfc_workspace_name,
+        'all|a'         => \$all_workspaces,
+        'quiet|q'       => \$quiet_mode,
+        'help|h'        => \$help
 ) or die $usage;
 
 die $usage if (!defined($tfc_org_name) || defined($help));
 die $usage if ( defined($tfc_workspace_name) &&  defined($all_workspaces));	# can't have both
 die $usage if (!defined($tfc_workspace_name) && !defined($all_workspaces));	# must have one
 
 if (! $ENV{ATLAS_TOKEN}) {
-	print STDERR "Terraform Cloud access token must be in ATLAS_TOKEN environment variable.\n";
-	die $usage;
+    my $error = "Terraform Cloud access token must be in ATLAS_TOKEN environment variable.";
+    error_to_sentry($error);
+    print STDERR "$error\n";
+    die $usage;
 }
 
 my $curl_header1 = "--header \"Authorization: Bearer $ENV{ATLAS_TOKEN}\"";
-my $curl_header2 = "--header \"Content-Type: application/vnd.api+json\"";
+my $curl_header2 = "--header \"Content-Type: application/vnd.api+json"";
 my $curl_headers = "$curl_header1 $curl_header2";
 if (defined($quiet_mode)) {
-	$curl_headers .= " --no-progress-meter";
+    $curl_headers .= " --no-progress-meter";
 }
 my $curl_query;
 my $curl_cmd;
 my $jq_cmd;
 my %workspace_list;
 if (defined($tfc_workspace_name)) {	# One workspace desired
 
-	# Get the workspace ID given the workspace name.
-
-	$curl_query = "\"https://app.terraform.io/api/v2/organizations/${tfc_org_name}/workspaces/${tfc_workspace_name}\"";
-	$curl_cmd   = "curl $curl_headers $curl_query";
-	$jq_cmd     = "jq '.data.id'";
-
-	$tfc_workspace_id = `$curl_cmd | $jq_cmd`;
-	$tfc_workspace_id =~ s/"//g;
-	chomp($tfc_workspace_id);
-
-	$workspace_list{$tfc_workspace_name} = $tfc_workspace_id;
+    # Get the workspace ID given the workspace name.
+    $curl_query = "\"https://app.terraform.io/api/v2/organizations/${tfc_org_name}/workspaces/${tfc_workspace_name}\"";
+    $curl_cmd   = "curl $curl_headers $curl_query";
+    $jq_cmd     = "jq '.data.id'";
+
+    system("$curl_cmd | $jq_cmd > /tmp/workspace_id.txt");
```

Comment:
It might be useful to make this a sub.

Reply:
I'm thinking slurp_output_from_command({command, e.g. "$curl_cmd | $jq_cmd"}, {file name});
Then the whole thing could be: $tfc_workspace_id = slurp_output_from_command("$curl_cmd | $jq_cmd", "workspace_id.txt"); -- The function would prepend the "/tmp/" and run the system command, and using File::Slurp's read_file slurp the contents into a variable to return. It would clean up the temporary file and then return the value. Then you could reuse the code below (93-103).